packages/layers/src/multiscale-image-layer/multiscale-image-layer.js

```diff
@@ -98,11 +142,22 @@ const MultiscaleImageLayer = class extends CompositeLayer {
          * return type, and optional throw for performance.
          */
         const tiles = await Promise.all(selections.map(getTile));
-
         const tile = {
-          data: tiles.map(d => d.data),
-          width: tiles[0].width,
-          height: tiles[0].height
+          data: tiles.map(d =>
+            clipper.clip({
+              data: d.data,
+              width: tiles[0].width,
+              height: tiles[0].height
+            })
+          ),
+          width:
+            clipper.width < tileSize && tiles[0].width === tileSize
+              ? clipper.width
+              : tiles[0].width,
+          height:
+            clipper.height < tileSize && tiles[0].height === tileSize
+              ? clipper.height
+              : tiles[0].height
```

Comment:
I generally am just having a hard time understanding the logic of this bit of code. Now the logic for clipping the data and clipping the width and height are separated, which makes it difficult to reason through.